Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1100,16 +1100,16 @@ class BrowserViewController: UIViewController {
             return
         }
 
+        // Fire the readability check. This is here and not in the pageShow event handler in ReaderMode.js anymore
+        // because that event wil not always fire due to unreliable page caching. This will either let us know that
+        // the currently loaded page can be turned into reading mode or if the page already is in reading mode. We
+        // ignore the result because we are being called back asynchronous when the readermode status changes.
+        webView.evaluateJavaScript("\(ReaderModeNamespace).checkReadability()", completionHandler: nil)
```

Comment:
Less confusing to always fire this, and moving it here makes the check on line 1110 simpler.